File: src/Entity/Attendance.php
<?php

namespace App\Entity;

use App\Repository\AttendanceRepository;
use Doctrine\DBAL\Types\Types;
use Doctrine\ORM\Mapping as ORM;

#[ORM\Entity(repositoryClass: AttendanceRepository::class)]
class Attendance
{
    #[ORM\Id]
    #[ORM\GeneratedValue]
    #[ORM\Column]
    private ?int $id = null;

    #[ORM\ManyToOne(inversedBy: 'attendances')]
    private ?session $session = null;

    #[ORM\Column]
    private ?bool $attended = null;

    #[ORM\Column(type: Types::DATETIME_MUTABLE, nullable: true)]
    private ?\DateTimeInterface $timestamp = null;

    #[ORM\ManyToOne(inversedBy: 'attendances')]
    private ?participantEvent $participantEvent = null;

    #[ORM\Column]
    private ?int $code = null;


    public function getId(): ?int
    {
        return $this->id;
    }

    public function getSession(): ?session
    {
        return $this->session;
    }

    public function setSession(?session $session): static
    {
        $this->session = $session;

        return $this;
    }

    public function isAttended(): ?bool
    {
        return $this->attended;
    }

    public function setAttended(bool $attended): static
    {
        $this->attended = $attended;

        return $this;
    }

    public function getTimestamp(): ?\DateTimeInterface
    {
        return $this->timestamp;
    }

    public function setTimestamp(?\DateTimeInterface $timestamp): static
    {
        $this->timestamp = $timestamp;

        return $this;
    }

    public function getParticipantEvent(): ?participantEvent
    {
        return $this->participantEvent;
    }

    public function setParticipantEvent(?participantEvent $participantEvent): static
    {
        $this->participantEvent = $participantEvent;

        return $this;
    }

    public function getCode(): ?int
    {
        return $this->code;
    }

    public function setCode(int $code): static
    {
        $this->code = $code;

        return $this;
    }

 
}

File: rt/src/main/java/pi2425/swappy_javafx/entities/Attendance.java
package pi2425.swappy_javafx.entities;

import java.time.LocalDateTime;

public class Attendance {

    private int id;
    private Session session;
    private Participant_event participant_event;
    private boolean attended;
    private int code;
    private LocalDateTime timestamp;
    public Attendance() {}
    public Attendance(LocalDateTime timestamp,int id, Session session, Participant_event participant_event, boolean attended, int code) {
        this.id = id;
        this.session = session;
        this.participant_event = participant_event;
        this.attended = attended;
        this.code = code;
        this.timestamp = timestamp;
    }

    public Attendance(Session session, Participant_event participant_event, boolean attended, int code, LocalDateTime timestamp) {
        this.session = session;
        this.participant_event = participant_event;
        this.attended = attended;
        this.code = code;
        this.timestamp = timestamp;
    }

    public int getId() {
        return id;
    }

    public Session getSession() {
        return session;
    }

    public void setSession(Session session) {
        this.session = session;
    }

    public Participant_event getParticipant_event() {
        return participant_event;
    }

    public void setParticipant_event(Participant_event participant_event) {
        this.participant_event = participant_event;
    }

    public boolean isAttended() {
        return attended;
    }

    public void setAttended(boolean attended) {
        this.attended = attended;
    }

    public int getCode() {
        return code;
    }

    public void setCode(int code) {
        this.code = code;
    }

    public LocalDateTime getTimestamp() {
        return timestamp;
    }

    public void setTimestamp(LocalDateTime timestamp) {
        this.timestamp = timestamp;
    }

    @Override
    public String toString() {
        return "Attendance{" +
                "id=" + id +
                ", session=" + session +
                ", participant_event=" + participant_event +
                ", attended=" + attended +
                ", code=" + code +
                ", timestamp=" + timestamp +
                '}';
    }
}
